applicationinsights/channel/AsynchronousSender.py

```diff
@@ -123,6 +129,14 @@ def _run(self):
                 # if we didn't get any items from the queue, we're done here
                 if len(data) == 0:
                     break
+                
+                # Drop extra events from the queue until it no larger than send_buffer_size.
+                # This is a necessary evil to prevent the backlog from bloating forever in bandwidth-limited environments
+                eventsToDrop = max(0, local_queue._queue.qsize() - self._send_buffer_size)
```

Comment:
Shouldn't it be max queue length (I believe this variable is defined as being 500)? I'm pointing out here that it's comparing to the current queue length rather than what the queue's limit is. This change would give the result that @lostromb described in the first reply. I was also pointing out that supplying this limit in the constructor of the queue will allow python to enforce this limit natively, making this check unnecessary.